### PR TITLE
refactor: rename `ScopedArray` -> `Array`

### LINKED ISSUE #3593: Rename `ScopedArray`
Similar to #3590 but should not be attempted before #3363 and #3589 etc.

### PR DESCRIPTION
**Pull request by jaschdoc**
Closes #3593

The old `Array` has been renamed to `UnscopedArray`

### CODE REVIEW

**Comment by jaschdoc** [comment]
This currently fails because of the community build where `Array` is not yet renamed. It should pass without the community build (everything is green locally)

**Comment by jaschdoc** [comment]
Currently failing on the community build since the `Array` syntax has not yet been updated `UnscopedArray`

**Comment by jaschdoc** [comment]
@magnus-madsen :)

**Comment by magnus-madsen** [comment]
A few things:

- Most importantly, I want to get rid of `UnscopedArray`. We should have very few of those and then the programmer should be forced to write `Array[t, Static]` to be very explicit about what it means

File: main/src/library/Prelude.flix [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ///
-/// Array[t] is a type alias for a scoped array with a global lifetime.
+/// UnscopedArray[t] is a type alias for an array with a global lifetime.
 ///
-pub type alias Array[t] = ScopedArray[t, false]
+pub type alias UnscopedArray[t] = Array[t, false]
```
Comment (magnus-madsen) [inline main/src/library/Prelude.flix:4]: This should be gone

**Comment by jaschdoc** [comment]
RFR

**Comment by jaschdoc** [comment]
@magnus-madsen RFR btw 😃

**Comment by magnus-madsen** [comment]
This is a big one. Ill have a look through. I already see some conflicts, sorry :/

**Comment by jaschdoc** [comment]
@magnus-madsen RFR (finally) :)

**Comment by magnus-madsen** [comment]
@mlutze Can you fix the community to build? :)

**Comment by magnus-madsen** [comment]
@stephentetley This PR renames ScopedArray[t, r] to just `Array[t, r]`. If you want the "old" array you have to write `Array[t, Static]`.

**Comment by magnus-madsen** [comment]
@jaschdoc Thanks for all your hard work :)

**Pull request merged**